Common/Extensions.cs

```diff
@@ -2875,5 +2875,20 @@ public static OptionRight Invert(this OptionRight right)
         {
             return ComparisonOperator.Compare(op, arg1, arg2);
         }
+
+        /// <summary>
+        /// Centralized logic used by the data feeds to determine if we should emit auxiliary base data points.
+        /// For equities we only want to emit split/dividends events for non internal and only for <see cref="TradeBar"/> configurations
+        /// this last part is because equities also have <see cref="QuoteBar"/> subscriptions.
+        /// </summary>
+        /// <remarks>The "TimeSliceFactory" does not allow for multiple dividends/splits per symbol in the same time slice
+        /// but we don't want to rely only on that to filter out duplicated aux data so we use this at the top of
+        /// our data enumerator stacks to define what subscription should emit this data.</remarks>
+        /// <remarks>History provider is never emitting auxiliary data points</remarks>
+        public static bool ShouldEmitAuxiliaryBaseData(this SubscriptionDataConfig config)
+        {
+            var typeFilter = (config.Type == typeof(TradeBar) || config.Type == typeof(Tick) && config.TickType == TickType.Trade);
+            return (!config.IsInternalFeed && typeFilter) || config.IsCustomData;
```

Comment:
I modified this only slightly to allow CustomData configurations to emit aux data always because it broke one of our regressions by blocking its aux data flow.

Reply:
Why isn't the `IsInternalFeed` check applied for custom data like before?
Suggest we add a comment, maybe here as remarks, about the _current_ dependency of the FF enumerators with the delisted event data point

Reply:
👍🏽